### PR TITLE
LPS-133968 Create Tests for Portal/Portlet TableReferenceDefinitions

### PR DESCRIPTION
**Pull request by gislayne-vitorino**
This is an update for: [LPS-133968](https://liferay.atlassian.net/browse/LPS-133968)

### CODE REVIEW

**Comment by gislayne-vitorino** [comment]
ci:test:publications

File: modules/apps/change-tracking/change-tracking-test/src/testIntegration/java/com/liferay/change/tracking/internal/change/tracking/test/PortletPreferencesTableReferenceDefinitionTest.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Override
+	protected CTModel<?> addCTModel() throws Exception {
+		Portlet portlet = _portletLocalService.fetchPortletById(
+			TestPropsValues.getCompanyId(), RandomTestUtil.randomString());
```
Comment (pei-jung) [inline modules/apps/change-tracking/change-tracking-test/src/testIntegration/java/com/liferay/change/tracking/internal/change/tracking/test/PortletPreferencesTableReferenceDefinitionTest.java:44]: Doesn't this just return null? Use a real portletId to fetch the portlet or simply use null in your service call.

File: modules/apps/change-tracking/change-tracking-test/src/testIntegration/java/com/liferay/change/tracking/internal/change/tracking/test/PortletPreferencesTableReferenceDefinitionTest.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+		return _portletPreferencesLocalService.addPortletPreferences(
+			TestPropsValues.getCompanyId(), group.getGroupId(),
+			PortletKeys.PREFS_OWNER_TYPE_LAYOUT, PortletKeys.PREFS_PLID_SHARED,
+			RandomTestUtil.randomString(), portlet,
```
Comment (pei-jung) [inline modules/apps/change-tracking/change-tracking-test/src/testIntegration/java/com/liferay/change/tracking/internal/change/tracking/test/PortletPreferencesTableReferenceDefinitionTest.java:49]: You can use TestPropsValues.getPlid() for the plid.

File: modules/apps/change-tracking/change-tracking-test/src/testIntegration/java/com/liferay/change/tracking/internal/change/tracking/test/PortletPreferenceValueTableReferenceDefinitionTest.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Override
+	protected CTModel<?> addCTModel() throws Exception {
+		PortletPreferenceValue portletPreferenceValue =
+			_portletPreferenceValueLocalService.createPortletPreferenceValue(
+				RandomTestUtil.randomLong());
+
+		return _portletPreferenceValueLocalService.addPortletPreferenceValue(
+			portletPreferenceValue);
```
Comment (pei-jung) [inline modules/apps/change-tracking/change-tracking-test/src/testIntegration/java/com/liferay/change/tracking/internal/change/tracking/test/PortletPreferenceValueTableReferenceDefinitionTest.java:44]: We need to populate the fields with some dummy data. It should follow the logic in https://github.com/liferay/liferay-portal/blob/master/portal-impl/src/com/liferay/portal/service/impl/PortletPreferencesLocalServiceImpl.java#L1045-L1059

**Review by pei-jung — CHANGES_REQUESTED** [review]
@gislayne-vitorino can you move the tests to `modules/apps/portlet-preferences/portlet-preferences-test`?

**Comment by gislayne-vitorino** [comment]
ci:test:sf

**Comment by gislayne-vitorino** [comment]
ci:test:publications

**Review by pei-jung — APPROVED** [review]

**Comment by pei-jung** [comment]
ci:forward

**Pull request closed without merging**